eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/LightningMessageTypes.scala

```diff
@@ -106,9 +106,15 @@ case class TxComplete(channelId: ByteVector32,
                       tlvStream: TlvStream[TxCompleteTlv] = TlvStream.empty) extends InteractiveTxConstructionMessage with HasChannelId
 
 case class TxSignatures(channelId: ByteVector32,
-                        txId: ByteVector32,
+                        txHash: ByteVector32,
                         witnesses: Seq[ScriptWitness],
-                        tlvStream: TlvStream[TxSignaturesTlv] = TlvStream.empty) extends InteractiveTxMessage with HasChannelId
+                        tlvStream: TlvStream[TxSignaturesTlv] = TlvStream.empty) extends InteractiveTxMessage with HasChannelId {
+  val txId: ByteVector32 = txHash.reverse
+}
+
+object TxSignatures {
+  def apply(channelId: ByteVector32, tx: Transaction, witnesses: Seq[ScriptWitness]): TxSignatures = TxSignatures(channelId, tx.hash, witnesses)
+}
```

Comment:
Nit: I understand the purpose of this, but it is still possible to footgun ourselves by calling the main constructor, and it makes the content of the case class less obvious from the caller perspective. I would instead instantiate by explicitly specifying the arg name `txHash = tx.txid`.

Reply:
> You can't make the constructor private or you will have problem at deserialization.

Right, indeed!

Anyway it's also really painful to make a `case class`'s `apply` method private. We can make the constructor private by just defining it as `case class TxSignatures private (...)` but that only makes `val txSigs = new TxSignatures()` private, not `val txSigs = TxSignatures()`.

All the ways I've found to work-around this are a bit ugly...so let's keep it that way and rely on tests to ensure we don't mess this up.